libs/core/src/test/java/org/opensearch/core/xcontent/filtering/FilterPathTests.java

```diff
@@ -369,4 +370,20 @@ public void testMultipleFilterPaths() {
         assertThat(filterPath.getSegment(), is(emptyString()));
         assertSame(filterPath, FilterPath.EMPTY);
     }
+
+    public void testCompileWithEmptyString() {
```

Comment:
Seems this newly added test case is duplicated with the existing test case: https://github.com/opensearch-project/OpenSearch/blob/14f1c43c108f378b13d109ade364216c082fb858/libs/core/src/test/java/org/opensearch/core/xcontent/filtering/FilterPathTests.java#L122.